OptimizelySDKShared/OptimizelySDKShared/OPTLYManagerBase.m

```diff
@@ -67,6 +69,29 @@ @interface OPTLYManagerBase()
 
 @implementation OPTLYManagerBase
 
+- (void)cleanUserProfileService:(NSArray<OPTLYExperiment> *)experiments {
```

Comment:
Can you also add some unit tests for this method, particularly around failure scenarios

Reply:
@mikeng13 done